### ISSUE TITLE
'No emulator images (avds) found' issue after 1.2 upgrade

### ISSUE THREAD

**Issue by justin-umesh** [meteor/meteor#5315]
1. Download desired System Image by running:
   /Users/apple/.meteor/android_bundle/android-sdk/tools/android sdk
   2. Create an AVD by running:
   /Users/apple/.meteor/android_bundle/android-sdk/tools/android avd
   HINT: For a faster emulator, use an Intel System Image and install the HAXM
   device driver
   at ChildProcess.exitCallback (/tools/utils/processes.js:137:23)
   at ChildProcess.emit (events.js:98:17)
   at Process.ChildProcess._handle.onexit (child_process.js:820:12)

**Comment by martijnwalraven** [comment]
I think the issue is that you have `ANDROID_HOME` set to ` /Users/apple/.meteor/android_bundle/android-sdk`, the location of the old Android bundle. If you haven't done so already, install the Android SDK by following the [instructions](https://github.com/meteor/meteor/wiki/Mobile-Development-Install:-Android-on-Mac), and either remove or change the existing `ANDROID_HOME`.

**Comment by justin-umesh** [comment]
@martijnwalraven

Thanks, I have already followed the steps and successfully installed and configured correctly. But the emulator not launching the app. it says 'Starting app on Android Emulator'.

**Comment by martijnwalraven** [comment]
And it keeps saying that, no error messages? Note that the Android emulator is fairly slow, and we also need to compile the app first, so starting up might take a while.

**Comment by justin-umesh** [comment]
Yes you are right. But in older versions of meteor it was working extremely good.

**Comment by martijnwalraven** [comment]
Do you mean it is working now, just slow? And slower than before?

**Comment by justin-umesh** [comment]
Yes it is working. Its very slow compare to earlier meteor versions.

**Comment by martijnwalraven** [comment]
It should be faster actually, definitely not slower. Are you sure you've installed HAXM acceleration (should be done automatically when configuring Android Studio)?

**Comment by justin-umesh** [comment]
yes i have installed HAXM although its very slow.

**Comment by justin-umesh** [comment]
I have understood and fixed the problem. after creation of AVD we need to launch the emulator manually then run the 'meteor run android' command it will be very faster..

Thanks for your support.
Bye :)

**Comment by martijnwalraven** [comment]
Glad to hear it is working for you, closing this issue!

**Comment by ashtynabell** [comment]
Hey @martijnwalraven, I don't think this should be closed as this issue still persists. After upgrading to 1.2, I removed the `android` platform from my Meteor project and installed the Android SDK via the aforementioned instructions. 

I set `ANDROID_HOME`, added `$ANDROID_HOME/tools` and `$ANDROID_HOME/platform-tools` to my PATH, but still no go. I even reset `ANDROID_HOME` and added the tools again with a direct text-edit to my `.bash_profile`. Still no go ... I still get this same output: 
```
=> Errors executing Cordova commands:

   While running Cordova app for platform Android with options --emulator:
   Error: Command failed: /Users/AABell/myDE/greenbook/.meteor/local/cordova-build/platforms/android/cordova/run --emulator

   /Users/AABell/myDE/greenbook/.meteor/local/cordova-build/platforms/android/cordova/node_modules/q/q.js:126
   throw e;
   ^
   ERROR : No emulator images (avds) found.
   1. Download desired System Image by running: /Users/AABell/Library/Android/sdk/tools/android sdk
   2. Create an AVD by running: /Users/AABell/Library/Android/sdk/tools/android avd
   HINT: For a faster emulator, use an Intel System Image and install the HAXM device driver
   at ChildProcess.exitCallback (/tools/utils/processes.js:137:23)
   at ChildProcess.emit (events.js:98:17)
   at Process.ChildProcess._handle.onexit (child_process.js:820:12)

ExitWithCode:1
```

I removed the `android` platform from my project and reset the `ANDROID_HOME` environment variable a few times. Still no luck. Not sure what the issue is.

**Comment by iboxgithub** [comment]
Same issue here

**Comment by iboxgithub** [comment]
I tried to solve it by adding in Android 5.1.1 
- "Intel x86 Atom System Image" --> KO
- "Intel x86 Atom_64 System Image" (my laptop is Ubuntu 14.04 LTS, 64bits) --> KO

I will try adding the Google API System images but I am only doing empirical actions here, the page to follow is clearly imcomplete, to much stuff don't work out of the box, could we re-open this issue?

**Comment by autodidacticon** [comment]
@iboxgithub You'll need to open the Android Virtual Device Manager and create a new AVD from one of the system images you downloaded. After the AVD is created, start it from the AVD main menu and once it is running, you should then be able to start your application.

**Comment by iboxgithub** [comment]
Thanks, I also found this solution and posted it on another thread, I forgot this one.

**Comment by gghez** [comment]
Added **Android 5.1.1 (API 22)** *Intel x86 Atom 64 System Image* which is not downloaded with default Android Studio installation. Use `android sdk` to open Android Studio for this purpose.

<img width="700" alt="screen shot 2016-01-02 at 01 36 14" src="https://cloud.githubusercontent.com/assets/6266247/12074462/c65512de-b155-11e5-9a40-7c9f97c2d72d.png">

Then configure this downloaded image ( @iboxgithub, you probably miss this step ) using `android avd` command.

<img width="511" alt="screen shot 2016-01-02 at 17 56 34" src="https://cloud.githubusercontent.com/assets/6266247/12075246/56f67c6e-b17a-11e5-8c61-31598c5ba375.png">

Please update official doc (https://github.com/meteor/meteor/wiki/Mobile-Development-Install:-Android-on-Mac) to integrate that requirement. By the way, it was much more easier by the past using `meteor install-sdk`.

**Comment by aido179** [comment]
Had identical issues on a fresh install on OS X El Capitan.
Can confirm gghez's solution works, however, I'd add on mac you must select the API in the SDK manager and click `show details` on the bottom right before you can see the specific AVD. I then ran the `android avd` and created a new device using the exact same details as the screen shot above.

**Comment by AnthonyAstige** [comment]
@gghez Thanks, those instructions got me functional again.

**Details**
I stopped developing on my machine for a month or so, then upgraded from Ubuntu 14 to 15.10 when I returned. I had some other issues which the [regular documentation](https://github.com/meteor/meteor/wiki/Mobile-Development-Install:-Android-on-Linux) got me through. The instructions above from @gghez were the final steps in making ```meter run android``` work for me again.
